packages/components/src/components/ContentfulContainer/ContentfulContainer.tsx

```diff
@@ -6,45 +6,32 @@ import { ContentfulContainerAsHyperlink } from './ContentfulContainerAsHyperlink
 import type { ContentfulContainerAsHyperlinkProps } from './ContentfulContainerAsHyperlink';
 import { combineClasses } from '../../utils/combineClasses';
 
-export const ContentfulContainer: React.FC<ContentfulContainerAsHyperlinkProps> = (
-  sectionProps
-) => {
-  const { className, editorMode, children } = sectionProps;
+export const ContentfulContainer: React.FC<ContentfulContainerAsHyperlinkProps> = (props) => {
+  const { className, editorMode, children, cfHyperlink } = props;
 
-  if (sectionProps.cfHyperlink) {
-    return (
-      <ContentfulContainerAsHyperlink
-        className={className}
-        editorMode={editorMode}
-        cfHyperlink={sectionProps.cfHyperlink}
-        cfOpenInNewTab={sectionProps.cfOpenInNewTab}
-        {...sectionProps}>
-        {children}
-      </ContentfulContainerAsHyperlink>
-    );
+  if (cfHyperlink) {
+    return <ContentfulContainerAsHyperlink {...props}>{children}</ContentfulContainerAsHyperlink>;
```

Comment:
this could lead to unintentionally spreading some props onto the component that don't belong, which could cause additional rendering and confusion. We've also ran into a few React errors/warnings about props being spread onto elements that don't accept them. Unless we are sure the child component takes the same props as the parent I would suggest manually adding each variable.

Reply:
This component was already spreading via `{...sectionProps}` before I refactored.  I think I'll switch these props to be individually passed in instead of spreading though, you raised some good points about spreading here

Reply:
I think I'm going to leave this one as-is and make a story for future refactoring the ContentfulContainer component code.

Spreading the props seems to work better with the way the `ContentfulContainerAsHyperlinkProps` type is currently defined conditionally based on `editorMode`.  I attempted to refactor this and pass in all the props explicitly without spreading but it becomes challenging with the conditional prop types, and there's tradeoffs to consider w/ refactoring the conditional prop type

Reply:
Created https://contentful.atlassian.net/browse/ALT-331 to capture some ideas for additional refactoring. Feel free to update this ticket with any other ideas!